model/internal/otlp_wrapper.go

```diff
@@ -196,3 +236,30 @@ func intExemplarToExemplar(src []otlpmetrics.IntExemplar) []otlpmetrics.Exemplar
 	}
 	return exemplars
 }
+
+func numberDataPointsLabelsToAttributes(dps []*otlpmetrics.NumberDataPoint) {
+	for i, _ := range dps {
+		if dps[i].Labels != nil {
+			dps[i].Attributes = labelsToAttributes(dps[i].Labels)
```

Comment:
This has been moved out of this PR, but I think it's an important point to clarify. Based on the comments in the proto, https://github.com/open-telemetry/opentelemetry-proto/blob/48ca003a72365f5591f1acd9b065ea8316be2ade/opentelemetry/proto/metrics/v1/metrics.proto#L307-L312 what are your thoughts on the right way to move forward? I suppose we can either:

1. assume the Attributes are empty and override them with labels
2. upsert Labels into Attributes
3. insert Labels that are not present only

3 seems to make the most sense to me, but it's not a strong preference.

Reply:
I would say we need to assume Attributes is empty:
Users cannot constructor custom protos since they are hidden. This case can happen only when we receive from the wire. So `MetricsCompatibilityChanges` is only called from data received on the wire which can be in new or old format but not a mix.

Depends how backwards compatible we want to be:
1. If only labels is set we upsert everything to the Attributes
2. If both are received the expectation is that they are the same, do nothing
3. If only attributes is received then do nothing.

On the sent side:
* If we want to do the correct thing we should translate attributes to labels and send both, but this is not what we did with IntSum/IntGauge. So maybe just a CHANGELOG entry and say that we broke this?

Reply:
Got it.

When receiving:
- we should check if both labels and attributes are set, if so, don't do anything
- if labels are set, upsert into attributes and leave labels as is. 
- don't do anything with the attributes only case.

When exporting, should labels always be dropped here? I worry that we would end up with inconsistent behaviour if labels are sometimes set and sometimes not.